### PR TITLE
Add developer setup instructions for React Native app

### PR DESCRIPTION
**Pull request by hmhealey**
Let me know if I screwed anything up with the rst format since it's the first time I've used it.

### CODE REVIEW

**Comment by it33** [comment]
Sweet! Thanks HH!

File: source/developer/mobile-developer-setup.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    - Add ``ANDROID_HOME/tools`` and ``ANDROID_HOME/platform-tools`` to the ``PATH``.
+
+2. Run ``android`` to open the Android SDK Manager and install the following packages:
+
+  - Tools > Android SDK Tools 25.2.5 or higher
+
+  - Tools > Android SDK Platform-tools 25.0.3
+
+  - Tools > Android SDK Build-tools 23.0.1
```
Comment (enahum) [inline source/developer/mobile-developer-setup.rst]: Also need Build Tools 25.0.2

File: source/developer/mobile-developer-setup.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  - Android 6.0 > SDK Platform 23
+
+  - Android 6.0 > Google APIs 23
+
+  - Android 5.1.1 > SDK Platform 22
+
+  - Android 5.1.1 > Google APIs 22
+
+  - Extras > Android Support Repository
```
Comment (enahum) [inline source/developer/mobile-developer-setup.rst]: Also needed Extras:
* Android Support Library
* Google Play services
* Google Repository

File: source/developer/mobile-developer-setup.rst (hunk 1 above)
Comment (hmhealey) [inline source/developer/mobile-developer-setup.rst]: Are you sure about these? The versions I have listed match what I have installed

File: source/developer/mobile-developer-setup.rst (hunk 2 above)
Comment (hmhealey) [inline source/developer/mobile-developer-setup.rst]: Same with these libraries. I don't have Google Play Services or the Google Repository

File: source/developer/mobile-developer-setup.rst (hunk 1 above)
Comment (enahum) [inline source/developer/mobile-developer-setup.rst]: Yes this one is needed for a dependency `react-native-device-info`

File: source/developer/mobile-developer-setup.rst (hunk 2 above)
Comment (enahum) [inline source/developer/mobile-developer-setup.rst]: This are to enable push notifications in the emulator

**Comment by hmhealey** [comment]
Ah I haven't updated my local instance to use either of those libraries. I'll update this

**Comment by hmhealey** [comment]
@enahum I updated the build tools and added the Google Play Services and Google Repository to be included

File: source/developer/mobile-developer-setup.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
+..  _mobile-developer-setup:
+
+Mobile Developer Machine Setup
+==================================
+
+The following instructions apply to new unreleased mobile apps for iOS and Android built in React Native. They are intended to release in Beta at the end of March 2017. Source code can be found at: https://github.com/mattermost/mattermost-mobile
```
Comment (lfbrock) [inline source/developer/mobile-developer-setup.rst]: @JeffSchering will this autolink once it's on docs site, or do we need to format it as a link?

Comment (JeffSchering) [inline source/developer/mobile-developer-setup.rst]: @lfbrock  It'll autolink.

**Pull request merged**